kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/LogReplaySuite.scala

```diff
@@ -298,4 +300,77 @@ class LogReplaySuite extends AnyFunSuite with TestUtils {
+      assert(snapshotImpl.getProtocol.getWriterFeatures.asScala.toList == List("domainMetadata",
+        "clustering", "v2Checkpoint"))
+      assert(snapshot.getVersion(defaultTableClient) == 3)
+
+      // Validate AddFiles from sidecars found against Spark connector.
+      val scan = snapshot.getScanBuilder(defaultTableClient).build()
+      val foundFiles =
+        collectScanFileRows(scan).map(InternalScanFileUtils.getAddFileStatus).map(
+          _.getPath.split('/').last).toSet
+       val expectedFiles = DeltaLog.forTable(
+        spark, path).update().allFiles.collect().map(_.path).toSet
+      assert(foundFiles == expectedFiles)
+    }
+  }
+
+  test("UUID named checkpoint with actions") {
+    // Get table path and create snapshot.
+    val path = goldenTablePath(s"checkpoint-uuid-with-actions")
+    val snapshot = latestSnapshot(path)
+    val snapshotImpl = snapshot.asInstanceOf[SnapshotImpl]
+
+    // Validate metadata/protocol loaded correctly from checkpoint manifest.
+    assert(snapshotImpl.getMetadata.getId == "testId")
+    assert(snapshotImpl.getProtocol.getMinReaderVersion == 3)
+    assert(snapshotImpl.getProtocol.getMinWriterVersion == 7)
+    assert(snapshotImpl.getProtocol.getReaderFeatures.asScala.toList == List("v2Checkpoint"))
+    assert(snapshotImpl.getProtocol.getWriterFeatures.asScala.toList == List("v2Checkpoint"))
+    assert(snapshot.getVersion(defaultTableClient) == 0)
+
+    // Validate AddFiles from sidecars found against Spark connector.
+    val scan = snapshot.getScanBuilder(defaultTableClient).build()
+    val foundFiles =
+      collectScanFileRows(scan).map(InternalScanFileUtils.getAddFileStatus).map(
+        _.getPath.split('/').last).toSet
+    assert(foundFiles == Set("addfile"))
+  }
+
+  test("compatibility checkpoint with sidecar files") {
+    // Get table path and create snapshot.
+    val path = goldenTablePath(s"checkpoint-v2-compatibility")
+    val snapshot = latestSnapshot(path)
+    val snapshotImpl = snapshot.asInstanceOf[SnapshotImpl]
+
+    // Validate metadata/protocol loaded correctly from checkpoint manifest.
+    assert(snapshotImpl.getMetadata.getId == "testId")
+    assert(snapshotImpl.getProtocol.getMinReaderVersion == 3)
+    assert(snapshotImpl.getProtocol.getMinWriterVersion == 7)
+    assert(snapshotImpl.getProtocol.getReaderFeatures.asScala.toList == List("v2Checkpoint"))
+    assert(snapshotImpl.getProtocol.getWriterFeatures.asScala.toList == List("v2Checkpoint"))
+    assert(snapshot.getVersion(defaultTableClient) == 3)
+
+    // Validate AddFiles from sidecars found against Spark connector.
+    val scan = snapshot.getScanBuilder(defaultTableClient).build()
+    val foundFiles =
+      collectScanFileRows(scan).map(InternalScanFileUtils.getAddFileStatus).map(
+        _.getPath.split('/').last).toSet
+    val expectedFiles = DeltaLog.forTable(
+      spark, path).update().allFiles.collect().map(_.path).toSet
+    assert(foundFiles == expectedFiles)
+  }
```

Comment:
A few more tests that would be good to have (maybe in a follow up PR if needed):
1. Multiple V2 Checkpoints at the same version.
2. Manifest with non-file actions other than metadata and protocol.
3. A table where a v2 checkpoint exists at version X but a classic checkpoint exists at version X + c.

Reply:
1. Added
2. Covered [here](https://github.com/delta-io/delta/pull/2826/files#diff-1004ccc9ef0014bf1b70f65d9a58f53fe8e3ce9ee543d8f7c711d69c052d388bR304): clustered tables have DomainMetadata actions.
3. Covered [here](https://github.com/delta-io/delta/pull/2826/files#diff-8685ce82b9bf552cb1a5e0e7d1aad80bea14ac7dbb7065e9610241439240accfR602)
4. Covered [here](https://github.com/delta-io/delta/pull/2826/files#diff-cffa92f50a43305e47e63051a7946b1e1a0cfa14e4ca9b75336c520ee815a581R494)

Reply:
How about some tests that also has `RemoveFile` actions later in the commits that remove files from `AddFile` in v2 checkpoint?